json/src/de.rs

```diff
@@ -699,14 +699,14 @@ impl<'a, R: Read> de::Deserializer for &'a mut Deserializer<R> {
     fn deserialize_bytes<V>(self, visitor: V) -> Result<V::Value>
         where V: de::Visitor
     {
-        try!(self.parse_whitespace());
-
-        let next = try!(self.peek());
-        let next = try!(next.ok_or(self.peek_error(ErrorCode::ExpectedSomeString)));
+        if try!(self.parse_whitespace()) { // true if eof
+            return Err(self.peek_error(ErrorCode::EOFWhileParsingValue));
```

Comment:
We don't need this check right? deserialize_enum and deserialize_option don't have it. If we hit EOF, the peek_or_null will return '\0' so it goes to self.deserialize(visitor) which handles this error.